dev/AppNotifications/AppNotificationUtility.cpp

```diff
@@ -232,61 +249,54 @@ void Microsoft::Windows::AppNotifications::Helpers::RegisterAssets(std::wstring
         &hKey,
         nullptr /* lpdwDisposition */));
 
-    // Retrieve DisplayName and IconUri
-    // - DisplayName: Retrieve from Shell. If not specified, fall back to filename.
-    // - Icon: Retrieve from Shell. If it's not the case or the file extension is unsupported, then throw.
-    winrt::com_ptr<IPropertyStore> propertyStore;
-    wil::unique_hwnd hWindow{ GetConsoleWindow() };
-    THROW_IF_FAILED(SHGetPropertyStoreForWindow(hWindow.get(), IID_PPV_ARGS(propertyStore.put())));
-
     // Retrieve the display name
     std::wstring displayName{};
+    displayName = SetDisplayNameBasedOnProcessName();
 
-    wil::unique_prop_variant propVariantDisplayName;
-    // Do not throw in case of failure. We can use the filepath approach below as fallback to set a DisplayName.
-    LOG_IF_FAILED(propertyStore->GetValue(PKEY_AppUserModel_RelaunchDisplayNameResource, &propVariantDisplayName));
-    if (propVariantDisplayName.vt == VT_LPWSTR && propVariantDisplayName.pwszVal != nullptr)
-    {
-        displayName = propVariantDisplayName.pwszVal;
-    }
+    // Retrieve the icon
+    std::wstring iconFilePath{};
 
-    if (displayName.length() == 0)
+    wil::unique_hwnd hWindow{ GetConsoleWindow() };
+
+    if (hWindow)
     {
-        THROW_IF_FAILED(wil::GetModuleFileNameExW(GetCurrentProcess(), nullptr, displayName));
+        // Retrieve DisplayName and IconUri
+        // - DisplayName: Retrieve from Shell. If not specified, fall back to filename.
+        // - Icon: Retrieve from Shell. If it's not the case or the file extension is unsupported, then throw.
+        winrt::com_ptr<IPropertyStore> propertyStore;
+        THROW_IF_FAILED(SHGetPropertyStoreForWindow(hWindow.get(), IID_PPV_ARGS(propertyStore.put())));
+
+        wil::unique_prop_variant propVariantDisplayName;
+        // Do not throw in case of failure. We can use the filepath approach below as fallback to set a DisplayName.
+        THROW_IF_FAILED(propertyStore->GetValue(PKEY_AppUserModel_RelaunchDisplayNameResource, &propVariantDisplayName));
+        if (propVariantDisplayName.vt == VT_LPWSTR && propVariantDisplayName.pwszVal != nullptr)
```

Comment:
Please check out the documentation for propvariant->Get()
https://docs.microsoft.com/en-us/previous-versions/windows/desktop/legacy/bb761473(v=vs.85)

You always check for VT_EMPTY after GetValue succeeds

Reply:
Not that displayName check for VT_EMPTY is best effort. If it isn't present, then we default to processname

Reply:
It can be either VT_LPWSTR or VT_EMPTY. The current check is sufficient.

Reply:
Oh I thought it was a flag?

Reply:
It is mostly a code saying what is inside the propVariant:
https://docs.microsoft.com/en-us/windows/win32/api/propidlbase/ns-propidlbase-propvariant?redirectedfrom=MSDN

Mostly, because there are a few flags indeed, but VT_EMPTY and VT_LPWSTR are mutually exclusive, for instance.